deployments/multiple-ragapps/docker-compose.yml

```diff
@@ -5,18 +5,66 @@ services:
       - 80:80
     labels:
       - traefik.enable=true
+      # API and dashboard
       - traefik.docker.network=ragapp-network
-      # Default username: admin, password: admin
-      - traefik.http.middlewares.admin-auth.basicauth.users=${USERNAME:-admin}:${HASHED_PASSWORD:-$$apr1$$3gG256rA$$BomHrC/uMhL807k/t7Wd6/}
-    volumes:
+      # Keycloak OpenID plugin configuration
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.KeycloakURL=http://localhost/auth
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.ClientID=ragapp
+      # Secret from the example ragapp keycloak realm
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.ClientSecret=jh4yDlJ5QsBSafVNaqqLxEG3vKcYyNgD 
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.KeycloakRealm=ragapp
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.Scope=openid
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.TokenCookieName=AUTH_TOKEN
+      - traefik.http.middlewares.my-keycloakopenid.plugin.keycloakopenid.UseAuthHeader=false
+    volumes:  
       - /var/run/docker.sock:/var/run/docker.sock:ro
+      - ./traefik/traefik.yml:/traefik_config.yml:ro
```

Comment:
can we (and does it make sense) to add the traefik labels into the config too?

Reply:
The traefik.yml is for static config, the other labels are dynamic config; i just let them here to easier update and they're just fine for now, If we keep adding more configurations, we can always move to a file later on. (https://doc.traefik.io/traefik/getting-started/configuration-overview/)

Reply:
ok, can't we set some defaults in the static config for each container? the label configs are quite long :D

Reply:
I tried to move config to file but cannot make it work well (due to the Traffik namespace suffix, the newly created app from manager won't work with predefined stuff like middlewares in the file config). I'll have a refactor on this later.